### ISSUE TITLE
tap-parser@1.3.x introduced regression

### ISSUE THREAD

**Issue by rmg** [tapjs/tapjs#281]
When tap-parser@1.3.0/1.3.1 was published it caused `tap` to start failing tests that were previously passing. The failures have no indication of why they are failing.

Example of the less than helpful output:

```
test/test-runner.js ................................. 11/12
  not ok test/test-runner.js
    timeout: 30000
    arguments:
      - test/test-runner.js
    command: /Users/ryan/.nvm/versions/node/v4.4.7/bin/node
    file: test/test-runner.js
    results:
      pass: 1
      plan:
        start: 1
        end: 1
      ok: false
      count: 1

total ............................................... 11/12


  11 passing (1s)
  1 failing
```

Installing tap-parser@1.2.x and then re-running the test results in the expected behaviour:

```
test/test-runner.js ................................. 11/11
total ............................................... 11/11

  11 passing (1s)

  ok
```

Ref: https://github.com/tapjs/tap-parser/issues/36#issuecomment-239056359

**Comment by isaacs** [comment]
Yeah, it's a bug. Forgot to revert this part. Will do shortly and publish.

**Comment by rmg** [comment]
@isaacs thanks. It kinda broke our CI out of the blue :-/

**Comment by isaacs** [comment]
Ugh, can't unpublish after 24 hours.  Curse those npm, Inc. folks and their excessively reasonable policies!

Published a tap 6.3.1 that depends on tap-parser 1.2.  Also published a tap-parser 1.3.2 which is a carbon-copy of 1.2.2, and deprecated the 1.3.0 and 1.3.1 versions.

> It kinda broke our CI out of the blue :-/

Yeah, it's a really strange bug that seems to be a combination of how tap was "cleverly" adding `Subtest: <name>` comments to child processes, and a race with the real output of the child process, which seems much more likely to strike on Linux than on OS X.  I have a clear understanding of the root cause, and plan to implement a forward-moving fix this weekend to get us to not-indented `Subtest` directives in a way that doesn't break existing users.
